sdk/mobile-center-distribute/src/main/java/com/microsoft/azure/mobile/distribute/Distribute.java

```diff
@@ -419,10 +420,17 @@ private synchronized void resumeDistributeWorkflow() {
             int downloadState = getStoredDownloadState();
             if (mReleaseDetails == null && downloadState != DOWNLOAD_STATE_COMPLETED) {
                 mReleaseDetails = DistributeUtils.loadCachedReleaseDetails();
+
+                /* If cached release is optional and we have network, we should not reuse it. */
+                if (mReleaseDetails != null && !mReleaseDetails.isMandatoryUpdate() &&
+                        NetworkStateHelper.getSharedInstance(mContext).isNetworkConnected() &&
```

Comment:
No this check is to avoid showing an old update dialog for optional update if I can get a more recent dialog. Remember we don't replace dialogs and wait for next time.